app/assets/javascripts/discourse/app/components/d-lightbox.js

```diff
@@ -0,0 +1,476 @@
+      this.hasLoadingError && `${base}--has-loading-error`,
+      this.willClose && `${base}--will-close`,
+      this.isRotated &&
+        this.rotationAmount &&
+        `${base}--is-rotated-${this.rotationAmount}`
+    );
+
+    return classNames.filter(Boolean).join(" ");
+  }
+
+  get shouldDisplayMainImageArrows() {
+    return (
+      !this.options.isMobile &&
+      this.canNavigate &&
+      !this.hasCarousel &&
+      !this.isZoomed &&
+      !this.isRotated
+    );
+  }
+
+  get shouldDisplayCarousel() {
+    return this.hasCarousel && !this.isZoomed && !this.isRotated;
+  }
+
+  get shouldDisplayCarouselArrows() {
+    return (
+      !this.options.isMobile &&
+      this.totalItemCount >= this.options.minCarosuelArrowItemCount
+    );
+  }
+
+  get shouldDisplayTitle() {
+    return !this.hasLoadingError && !this.isZoomed && !this.isRotated;
+  }
+
+  get totalItemCount() {
+    return this.items?.length || 0;
+  }
+
+  get counterIndex() {
+    return this.currentIndex ? this.currentIndex + 1 : 1;
+  }
+
+  get canNavigate() {
+    return this.items?.length > 1;
+  }
+
+  get canZoom() {
+    return !this.hasLoadingError && this.currentItem?.canZoom;
+  }
+
+  get canRotate() {
+    return !this.hasLoadingError;
+  }
+
+  get canDownload() {
+    return !this.hasLoadingError && this.options.canDownload;
+  }
+
+  get canFullscreen() {
+    return !this.hasLoadingError;
+  }
+
+  get hasLoadingError() {
+    return this.currentItem?.hasLoadingError;
+  }
+
+  get nextButtonIcon() {
+    return this.options.isRTL ? "chevron-left" : "chevron-right";
+  }
+
+  get previousButtonIcon() {
+    return this.options.isRTL ? "chevron-right" : "chevron-left";
+  }
+
+  get zoomButtonIcon() {
+    return this.isZoomed ? "search-minus" : "search-plus";
+  }
+
+  @bind
+  async registerAppEventListeners() {
+    this.appEvents.on(LIGHTBOX_APP_EVENT_NAMES.OPEN, this.open);
+    this.appEvents.on(LIGHTBOX_APP_EVENT_NAMES.CLOSE, this.close);
+  }
+
+  @bind
+  async deregisterAppEventListners() {
+    this.appEvents.off(LIGHTBOX_APP_EVENT_NAMES.OPEN, this.open);
+    this.appEvents.off(LIGHTBOX_APP_EVENT_NAMES.CLOSE, this.close);
+  }
+
+  @bind
+  async open({ items, startingIndex, callbacks, options }) {
+    this.options = options;
+
+    this.items = items;
+    this.currentIndex = startingIndex;
+    this.callbacks = callbacks;
+
+    this.isLoading = true;
+    this.isVisible = true;
+
+    await this.#setCurrentItem(this.currentIndex);
+
+    if (
+      this.options.zoomOnOpen &&
+      this.currentItem?.canZoom &&
+      !this.currentItem?.isZoomed
+    ) {
+      this.toggleZoom();
+    }
+
+    this.callbacks.onOpen?.({
+      items: this.items,
+      currentItem: this.currentItem,
+    });
+  }
+
+  @bind
+  async close() {
+    this.willClose = true;
+
+    discourseLater(this.cleanup, this.animationDuration);
+
+    this.callbacks.onClose?.();
+  }
+
+  async #setCurrentItem(index) {
+    await this.#onBeforeItemChange();
```

Comment:
we are not waiting on anything inside #onBeforeItemChange so it doesn’t need to await here, and also doesn’t need to be `async #onBeforeItemChange`

Reply:
Yes you right, I missed that one, that seems correct